src/v/cluster/rm_stm.cc

```diff
@@ -2836,9 +2826,7 @@ void rm_stm::spawn_background_clean_for_pids(rm_stm::clear_type type) {
         break;
     }
     case rm_stm::clear_type::idempotent_pids: {
-        if (
-          _log_state.lru_idempotent_pids.size()
-          <= _max_concurrent_producer_ids()) {
+        if (_log_state.seq_table.size() <= _max_concurrent_producer_ids()) {
```

Comment:
seq_table contains not only idempotent pids.
We can not use it to check the size of idempotent_pids. 
For example _max_concurrent_producer_ids =n + 1. You can have n + 1 transaction pids and 1 idempotent_pid. So you will clear idempotent_pid in this case

Reply:
Good point, I thought is_idempotent() is a mutually exclusive from is_transactional(), looks like its not. I'm testing another version of the fix, will get back.